src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -106,7 +102,7 @@ protected function doHeloCommand(): void
 
         /** @var SocketStream $stream */
         $stream = $this->getStream();
-        if (!$stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
+        if ($stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
```

Comment:
This change is wrong and must be reverted. If TLS is disabled AND the server supports STARTTLS, then we enable it.

Reply:
I don't know how, But without this patch, it makes the server crash. 
How could I disable TLS? (if I know the server does not support it)

Reply:
I really don't know enough about TLS. Do you want a simple reproducer?
(I can work a bit on this topic too)